### ISSUE TITLE
Can't fetch from local repo

### ISSUE THREAD

**Issue by devharsh** [github/git-lfs#1207]
I had created a local repo on my windows 7 pc
C:\Users\UserName\Original

Then I cloned it in
C:\Users\UserName\Clone

using git clone ../Original command.

Now I have LFS pointers created inside my clone under directory
C:\Users\UserName\Clone\Original

But when I try git lfs fetch it throws these errors:

Post c:///Users/UserName/Original.git/info/lfs/objects/batch: unsupported protocol
 scheme "c"
Post c:///Users/UserName/Original.git/info/lfs/objects/batch: unsupported protocol
 scheme "c"

**Comment by technoweenie** [comment]
Unfortunately, LFS requires a server to work. This way you don't have to keep all of the files on every local clone, as repositories using LFS can potentially get too large to work with.

There are two potential workarounds:

1. Use [Git worktrees](https://git-scm.com/docs/git-worktree). This lets you checkout separate branches of a single git repository into separate working directories.
2. If you really want to clone a local repository, try using the `--reference` flag: `git clone --reference Original https://git-server.com/devharsh/original Clone`. It's not quite the same as what you're doing. It's still cloning the remote repository into the `Clone` directory, but it will use `C:\Users\UserName\Original\.git\lfs` so it doesn't have to re-download the files.

**Comment by devharsh** [comment]
Thanks for the quick response.. I will look into it..

**Comment by sophana** [comment]
I didn't know about this. Even on linux, there is the same problem.
`unsupported protocol scheme ""`

So adding lfs files feature makes git centralized?
Couldn't we have a feature that would copy the files for empty protocol scheme ""
Adding the ssh protocol would be nice also...

**Comment by technoweenie** [comment]
> So adding lfs files feature makes git centralized?

Right now, yes. It's possible that this could change in the future. But we're focusing on getting the core user experience right before branching out into different protocols.

> Adding the ssh protocol would be nice also...

There is support for SSH remotes if the server has a custom `git-lfs-authenticate` command. But the file transfers don't flow through SSH.

**Comment by minrk** [comment]
@technoweenie thanks for the pointer to `--reference`! That helps my case. Since the local repo contains all of the information in that command, is there any chance of lfs being able to do this automatically (i.e. if source is local, lookup source's origin for lfs)?

**Comment by technoweenie** [comment]
> is there any chance of lfs being able to do this automatically (i.e. if source is local, lookup source's origin for lfs)?

Yes, there's a chance. It's just not a high priority right now. We're open to accepting a contribution from someone though.

**Comment by Weslo** [comment]
So I have some copies of a repo that push/pull from a local original copy. They are used to maintain separate cached data for different build targets:

project/root/origin <-> remote
project/root/a <-> project/root/origin
project/root/b <-> project/root/origin

Right now I have some scripts that pull from remote once into origin, then pull origin into a and b. How can I support LFS here? Do I need to delete and git clone --reference a and b each time origin finds LFS changes, since LFS doesn't support operations against a local origin?

**Comment by ttaylorr** [comment]
> Do I need to delete and git clone --reference a and b each time origin finds LFS changes, since LFS doesn't support operations against a local origin?

If I'm understanding your question correctly, simply `cp`-ing from the root's .git/lfs/object's cache should be sufficient.

**Comment by Weslo** [comment]
@ttaylorr Thank you, I'll try this if my current approach doesn't work out. For the time being, I'm skipping smudges and swapping remotes between the git pull and git lfs pull calls. So the normal git objects are fetched from the project/root/origin, but a and b pull lfs objects from the server.
